docs/Behaviour - Client Side.md

```diff
@@ -2,16 +2,47 @@
 
 ## Active Constraints of Sender
 
-Before making an [IS-05][IS-05] connection, NMOS Controller chooses a Sender and a number of Receivers. Then NMOS Controller MUST `GET /constraints/supported` from the Sender, MUST collect Receiver Capabilities from the Receivers and make such processing of them that they can be utilized for building Active Constraints satisfying all (by default) or some (depending on user preferences) of the Receivers and using Parameter Constraint URNs supported by the Sender. If the Sender supports fewer Parameter Constraint URNs than used in the Receiver Capabilities, the NMOS Controller SHOULD inform the user about it. If the processing of the Receiver Capabilities results in an empty `constraint_sets` of Active Constraints, NMOS Controller SHOULD inform the user about it. After that NMOS Controller MUST `PUT /constraints/active` to the Sender and SHOULD make the IS-05 connections if the Constraints have been applied succesfully to the Sender. After breaking these connections via IS-05, NMOS Controller is RECOMMENDED to `DELETE /constraints/active` of the Sender.
+An NMOS Controller manages stream compatibility using the Active Constraints of the Sender.
+The Active Constraints of the Sender are expressed as Constraint Sets that restrict the stream the Sender can transmit.
+The Controller MAY build these Constraint Sets from the Receiver Capabilities of one or more Receivers.
 
-Constraint Sets of Active Constraints SHOULD represent a consensus among Receivers. For example, there are Receiver A, Receiver B, Receiver C with equal Receiver Capabilities consisting of Constraint Sets 1, 2, 3, 4, 5 and Receiver D with Constraint Sets 2, 3, 4, 5 and 6 which is incompatible with the previous ones. In this case the consensus is Constraint Sets 2, 3, 4, 5 which are common for all the four Receivers. The user may control how the consensus is obtained, providing preferences to the NMOS Controller among the Receivers and the Constraint Sets of each Receiver and criteria about the size of the consensus.
+### Building the Constraint Sets
 
-`urn:x-nmos:cap:meta:preference` property in Constraint Sets of Receiver Capabilities indicates relative preference between Constraint Sets of that Receiver. The NMOS Controller MAY use these values to make a decision about how to process these Constraint Sets and determine preference of Constraint Sets of Active Constraints. It SHOULD NOT propagate values of this property as is from Receiver Capabilities to Active Constraints.
+The NMOS Controller MUST `GET /constraints/supported` from the Sender, MUST collect Receiver Capabilities from the Receivers and process them to build Constraint Sets satisfying all (by default) or some (depending on User preferences) of the Receivers, using Parameter Constraints supported by the Sender.
```

Comment:
... still describing only one way of getting active constraints "MUST collect Receiver Capabilities from the Receivers" even if the previous paragraph indicates "The Controller MAY build these Constraint Sets from the Receiver Capabilities of one or more Receivers." which open the possibility of build constraint sets some other way.

It seems to me that we must have something like "constraint sets can be build from Receiver capabilities or from some Controller configurations or user requirements expressed as BCP-004-01 capabilities ..."

Reply:
Agreeing on this one. Otherwise we are assuming that having Receivers is mandatory, which excludes the case with changing configuration of the Sender without any Receivers intact.

Reply:
You're absolutely right, having a MUST following on from a MAY is bad spec language. I'll rework it.

Reply:
@SoloAs, @alabou, I've had another go...